iothub_client/src/iothub_client_properties.c

```diff
@@ -69,137 +71,159 @@ static void AdvanceCountersAfterWrite(size_t currentOutputBytes, char** currentW
     }
 }
 
+// To correctly check snprintf for success, we need to check both it's not negative AND 
+// that the (number of bytes written ) != (sizeof buffer).
+static int PropertySnprintf(char* buffer, size_t count, const char* format, ...)
+{
+    va_list arg_list;
+    va_start(arg_list, format);
+
+    int currentOutputBytes = vsnprintf(buffer, count, format, arg_list);
+    if ((currentOutputBytes < 0) || (currentOutputBytes == (int)count))
```

Comment:
Nit:  Would it be more complete to have `currentOutputBytes >= (int)count` ?
Valid range is (0, currentOutputBytes), not (0, currentOutputBytes) U (currentOutputBytes, max positive int]

Reply:
RE differentiating the error - no, I really don't care how I got there just the bytes I wanted to come out didn't come and then to immediatlley cascade error up.

I think code as written is fine, since it captures the two error cases of "-1" and buffer getting terminated we'd actually hit and is a bit closer to other samples I've seen of this being done in past.